androidsvg/src/main/java/com/caverock/androidsvg/SVGImageView.java

```diff
@@ -25,9 +38,71 @@ public void setSVG(SVG svg) {
             throw new IllegalArgumentException("SVG can not be null");
         }
 
+        this.svg = svg;
+
         Bitmap bitmap = Bitmap.createBitmap((int) Math.ceil(svg.getDocumentWidth()), (int) Math.ceil(svg.getDocumentHeight()), Bitmap.Config.ARGB_8888);
         Canvas canvas = new Canvas(bitmap);
         svg.renderToCanvas(canvas);
         setImageBitmap(bitmap);
     }
+
+    public void setSVGTouchListener(SVGTouchListener listener) {
+        svgTouchlistener = listener;
+        gestureDetector = new GestureDetector(getContext(), this);
+        setOnTouchListener(this);
+    }
+
+    @Override
+    public boolean onDown(MotionEvent e) {
+        return false;
+    }
+
+    @Override
+    public void onShowPress(MotionEvent e) {
+
+    }
+
+    @Override
+    public boolean onSingleTapUp(MotionEvent e) {
```

Comment:
Let's extract the touch logic out into a separate inner class that extends SimpleOnGestureListener (so we don't have to implement all these methods that we're currently not using).